### PR TITLE
ANSA-288: Added logic gate evaluation algorithm to skipLogic checking

### PR DESCRIPTION
**Pull request by kaweesi**
this supports 'or' and 'and' logic gates evaluation at checks level instead of skipLogic function.

### CODE REVIEW

File: src/utils/skipLogic.js [hunk 1]
```diff
@@ -40,8 +40,12 @@ module.exports = function () {
     if (formField.skipLogic.checks.length > 0) {
       for (var i = 0; i < formField.skipLogic.checks.length; i++) {
         var check = formField.skipLogic.checks[i]
-        scope.$watch(check.variable, function (value, oldValue) {
-          skipLogicOperandCheck(scope, value, check)
+        scope.$watch(check.variable.startsWith('form.') ? check.variable : 'form.' + check.variable + '.$modelValue', function (value, oldValue) {
```
Comment (kaweesi) [inline src/utils/skipLogic.js]: just incase the parameter is passed in by name, this should handle it

Comment (rcrichton) [reply src/utils/skipLogic.js]: Wouldn't we also have to add watchers for all variables in a group. For groups to work correctly we need to watch each field that is part of the check. I don't think that is covered here.

File: test/utils/skipLogic.js [hunk 2]
```diff
@@ -291,3 +292,47 @@ tap.test('.skipLogic()', { autoend: true }, (t) => {
...
+    }
+
+    skipLogicGroupCheck(scope, 'yes', check)
+    t.notOk(scope.field.show)
+
+    check.logicGate = 'or'
+    skipLogicGroupCheck(scope, 'yes', check)
+    t.ok(scope.field.show)
+    skipLogicGroupCheck(scope, 'no', check)
+    t.ok(scope.field.show)
```
Comment (rcrichton) [inline test/utils/skipLogic.js]: We are missing a test case for when 'or' comes out as false.

Comment (kaweesi) [reply test/utils/skipLogic.js]: see below, lines 339 to 343

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "md-form-builder",
-  "version": "0.7.0",
+  "version": "0.8.0",
```
Comment (tmvumbi2) [inline package.json]: @kaweesi may be 0.9.0? @rcrichton  published a minor change an hour ago (currently 0.8.0)

**Pull request merged**